.clang-tidy

```diff
@@ -26,9 +26,18 @@ Checks:     '
             performance-*,
             portability-*,
             readability-*,
+            -bugprone-exception-escape,
+            -bugprone-macro-parentheses,
+            -bugprone-too-small-loop-variable,
+            -clang-analyzer-cplusplus.NewDelete,
+            -clang-analyzer-cplusplus.NewDeleteLeaks,
             -google-readability-braces-around-statements,
-            -google-readability-namespace-comments,
+            -modernize-use-nodiscard,
+            -modernize-avoid-c-arrays,
+            -performance-unnecessary-value-param,
```

Comment:
This actually found some good issues that I fixed, however it also blows up all over plan nodes with how shared_ptrs are used. Suggest reenabling once shared_ptrs are done being ripped out (@lmwnshn).

Reply:
Leave a comment?